app/police-custody-form/controllers.js

```diff
@@ -4,7 +4,124 @@ exports.addEvents = async function (req, res) {
   const user = req.user
   const move = await req.services.move.getById(moveId)
 
+  if (lockoutEvents.events === undefined) {
+    req.session.showErrorsSummary = true
+    return res.redirect(`move/${moveId}/police-custody-form`)
+  }
+
+  const submittedLockoutEvents = []
+  const errors = []
+
+  submittedLockoutEvents.push(lockoutEvents.events)
+
+  submittedLockoutEvents.flat().forEach(event => {
+    if (lockoutEvents[event] === '') {
+      errors.push(event)
+    }
+  })
+
+  if (errors.length > 0) {
+    const mappedErrors = mapErrorMessages(errors)
+
+    req.session.showErrorsSummary = true
+    req.session.errors = mappedErrors
+    req.session.formData = req.body
+
+    return res.redirect(`move/${moveId}/police-custody-form`)
+  }
+
   await req.services.event.postEvents(lockoutEvents, move, user)
 
-  res.redirect(`move/${moveId}/timeline`)
+  return res.redirect(`move/${moveId}/timeline`)
+}
+
+function mapErrorMessages(errors) {
+  const mappedErrorsWithMessages = []
+
+  errors.forEach(error => {
+    switch (error) {
```

Comment:
I think an object would work better here, to cut down on repeated code, something like ```
const EVENT_ERROR_MESSAGES = {
  PerViolentDangerous: 'Enter details of any violence that has taken place'
}
```
at the top of the file and then

```
const message = EVENT_ERROR_MESSAGES[error]
if (!message) return

mappedErrorsWithMessages.push({ error, message })
```

Reply:
Ohh yeah I was meant to refactor this. I like your idea, much cleaner ⭐